eo-maven-plugin/src/main/java/org/eolang/maven/MjParse.java

```diff
@@ -145,15 +148,30 @@ private Node parsed(final Path source, final String identifier) throws IOExcepti
             identifier, this.sourcesDir.toPath().relativize(source.toAbsolutePath()), xmir
         );
         final String name = new ObjectName(xmir).get();
+        final Node document = xmir.inner();
         if (!name.equals(identifier)) {
-            throw new IllegalArgumentException(
-                String.format(
-                    "Tojo identifier '%s' does not match to result object name '%s'",
-                    identifier, name
-                )
-            );
+            try {
+                new Xembler(
+                    new Directives()
+                        .xpath("/object")
+                        .addIf("errors")
+                        .add("error")
+                        .attr("check", "validate-object-name")
+                        .attr("severity", "critical")
+                        .set(
+                            String.format(
+                                "Tojo identifier '%s' does not match to result object name '%s'",
+                                identifier, name
+                            )
+                        )
+                ).apply(document);
```

Comment:
@h1alexbel you can use `applyQuietly()` with the same result, but less lines of code